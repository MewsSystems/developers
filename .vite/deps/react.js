import {
  require_react
} from "./chunk-32E4H3EV.js";
import "./chunk-G3PMV62Z.js";
export default require_react();
//# sourceMappingURL=react.js.map
