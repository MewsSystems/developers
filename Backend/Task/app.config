<?xml version="1.0" encoding="utf-8"?>
<configuration>
<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.7.1"/></startup>
  <appSettings>
    <add key="url" value="https://www.cnb.cz/cs/financni-trhy/devizovy-trh/kurzy-devizoveho-trhu/kurzy-devizoveho-trhu/denni_kurz.txt" />
    <add key="defaultCurrency" value="CZK" />
    <add key="decimalFormatProvider" value="cs-CZ" />
    <add key="lineSeparator" value="&#xA;" /> <!-- xml char for \n -->
    <add key="valueSeparator" value="|" />
    <add key="skippedRows" value="1" />
    <add key="rateColumnIndex" value="4" />
    <add key="quantityColumnIndex" value="2" />
    <add key="targetCurrencyColumnIndex" value="3" />
    <add key="expectedHeader" value="země|měna|množství|kód|kurz" />
  </appSettings>
</configuration>
