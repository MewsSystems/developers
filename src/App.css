:root {
  --btn-primary: #ff5a65;
  --btn-secondary: #ff5a6580;
  --black:#1c1d24;
  --text: #494951;
  --text-light: #8d8d92;
  --deep-blue: #172135;
  --ligth-blue: #edf4fe;
  --footer:#1c1d24;
  --sand: #f8f5f2;
  --white: #fff;
}

@font-face {
  font-family: 'Axiforma';
  src: local('Axiforma'), local('Axiforma'),
      url('./assets/axiforma/Axiforma-Regular.woff2') format('woff2'),
      url('./assets/axiforma/Axiforma-Regular.woff') format('woff'),
      url('./assets/axiforma/Axiforma-Regular.ttf') format('truetype');
}

@font-face {
  font-family: 'Axiforma-Bold';
  src: local('Axiforma'), local('Axiforma'),
      url('./assets/axiforma/Axiforma-Bold.woff2') format('woff2'),
      url('./assets/axiforma/Axiforma-Bold.woff') format('woff'),
      url('./assets/axiforma/Axiforma-Bold.ttf') format('truetype');
}

@font-face {
  font-family: 'Axiforma-Light';
  src: local('Axiforma'), local('Axiforma'),
      url('./assets/axiforma/Axiforma-Light.woff2') format('woff2'),
      url('./assets/axiforma/Axiforma-Light.woff') format('woff'),
      url('./assets/axiforma/Axiforma-Light.ttf') format('truetype');
}

html, body{
  font-family: "Axiforma", sans-serif;
  font-style: normal;
  color: var(--text);
}

.App {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
}

