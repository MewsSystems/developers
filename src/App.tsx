import { Form } from "./components/Form/Form";
import { Header } from "./components/Header/Header";
import styled from "styled-components";

const AppStyled = styled.div`
  text-align: center;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: start;
`;

function App() {
  return (
    <AppStyled>
      <Header />
      <Form />
    </AppStyled>
  );
}

export default App;
