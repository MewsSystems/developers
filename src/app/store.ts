import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import moviesReducer from '../features/movies/moviesSlice';
import movieDetailReducer from '../features/movieDetail/movieDetailSlice';

export const store = configureStore({
  reducer: {
    movies: moviesReducer,
    movieDetail: movieDetailReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
