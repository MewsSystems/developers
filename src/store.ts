import { combineReducers, configureStore } from '@reduxjs/toolkit';

import searchReducer from './reducers/searchReducer';
import resultsReducer from './reducers/resultsReducer';
import thunkMiddleware from './middlewares/thunkMiddleware';
import { throttle } from 'lodash';
import { loadState, saveState } from './utils/sessionStorage';
import detailReducer from './reducers/detailReducer';

const persistedState = loadState();

const store = configureStore({
  reducer: { search: searchReducer, results: resultsReducer, detail: detailReducer },
  preloadedState: persistedState,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(thunkMiddleware),
});

store.subscribe(
  throttle(() => {
    const state = store.getState();
    saveState(state);
  }, 500),
);

export default store;
