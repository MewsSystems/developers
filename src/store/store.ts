import { createStore } from "redux";
import {
  SET_GENRES,
  SET_SELECTED_MOVIE,
  SET_SELECTED_MOVIE_GENRES
} from "../models/actions";
import { InitialState } from "../models/initialState";

const initialState: InitialState = {
  genres: [],
  selectedMovie: null,
  selectedMovieGenres: []
};

const counterReducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_GENRES:
      return { ...state, genres: action.value };
    case SET_SELECTED_MOVIE:
      return { ...state, selectedMovie: action.value };
    case SET_SELECTED_MOVIE_GENRES:
      return { ...state, selectedMovieGenres: action.value };
    default:
      break;
  }
  return state;
};

const store = createStore(counterReducer);

export default store;
