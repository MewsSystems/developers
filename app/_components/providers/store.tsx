"use client";

import { FC, ReactNode } from "react";
import { Provider } from "react-redux";
import store from "@/store";

type Props = {
  children: ReactNode;
};

export const StoreProvider: FC<Props> = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
