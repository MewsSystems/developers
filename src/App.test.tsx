import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './app/store';
import App from './App';

test('render App.tsx', () => {
  const { container } = render(
    <Provider store={store}>
      <App />
    </Provider>
  );

  expect(container.innerHTML).toMatchSnapshot();
});
