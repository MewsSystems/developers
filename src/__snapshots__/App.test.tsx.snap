// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App.tsx 1`] = `"<div class=\\"App\\"><header class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-120r8wl-MuiPaper-root-MuiAppBar-root\\"><div class=\\"MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root\\"><div class=\\"MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root\\"><div class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1083xa6-MuiTypography-root\\">Movie search app</div><div class=\\"MuiBox-root css-1lvtzne\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"account of current user\\" aria-controls=\\"menu-appbar\\" aria-haspopup=\\"true\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"MenuIcon\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div><div class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-h7cdpm-MuiTypography-root\\">Movie search app</div><div class=\\"MuiBox-root css-1t6c9ts\\"><a class=\\"MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1q39md6-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" href=\\"/\\">Home<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></a></div></div></div></header><div class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-xwlqn3-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\" style=\\"text-align: center;\\"><div class=\\"MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root\\">Search movie</div><div class=\\"MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root\\">Search for a movie using input below</div></div><div class=\\"MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root\\" style=\\"width: 400px;\\"><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root\\"><span class=\\"notranslate\\">​</span><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"SearchIcon\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg></div><input id=\\"search-input\\" placeholder=\\"Search...\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-nnbavb\\"><span class=\\"notranslate\\">​</span></legend></fieldset></div></div><div class=\\"MuiGrid-root MuiGrid-item css-r4jj1p-MuiGrid-root\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-9ny27p-MuiGrid-root\\"></div></div></div></div></div>"`;
